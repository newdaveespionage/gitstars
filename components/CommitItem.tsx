import { DateTime } from 'luxon'

type CommitItemProps = {
  id: string
  name: string
  date: string
  message: string
}

const CommitItem = ({
  id,
  name,
  date,
  message,
}: CommitItemProps): JSX.Element => (
  <div key={id} className="card">
    <div className="commit-header">
      <span className="name">{name}</span>
      <span className="date">
        {DateTime.fromISO(date).toLocaleString(DateTime.DATETIME_SHORT)}
      </span>
    </div>
    <div className="message">{message}</div>
    <style jsx>{`
      .commit-header {
        display: flex;
        justify-content: space-between;
        margin-bottom: 0.5rem;
      }
      .name,
      .date,
      .message {
        line-height: 1.5rem;
      }
      .name {
        font-weight: bold;
      }
      .message {
        font-size: 0.9rem;
      }
    `}</style>
  </div>
)

export default CommitItem
