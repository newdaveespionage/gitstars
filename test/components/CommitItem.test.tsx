import React from 'react'
import { render } from '../testUtils'
import CommitItem from '../../components/CommitItem'

export const testCommitItemProps = {
  id: 'testid',
  name: 'Test Repo Name',
  date: '2021-04-19T03:57:00.279Z',
  message: 'refactor: creating snapshots',
}

describe('CommitItem', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<CommitItem {...testCommitItemProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
