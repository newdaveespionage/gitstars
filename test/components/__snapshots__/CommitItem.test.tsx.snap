// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitItem matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-2379951421 card"
  >
    <div
      class="jsx-2379951421 commit-header"
    >
      <span
        class="jsx-2379951421 name"
      >
        Test Repo Name
      </span>
      <span
        class="jsx-2379951421 date"
      >
        4/18/2021, 8:57 PM
      </span>
    </div>
    <div
      class="jsx-2379951421 message"
    >
      refactor: creating snapshots
    </div>
  </div>
</DocumentFragment>
`;
